### PR TITLE
Validator - Bounding Volume Spatial Coherence - Sphere Inside Box

### PR DESCRIPTION
**Pull request by rms13**
Issue #25 - sphere inside box

Added check for sphere type content bounding volume inside tile with box type bounding volume.

Creating 6 planes representing faces of the tile bounding box, and checking if sphere is on the wrong side or intersecting any of them.

### CODE REVIEW

File: validator/lib/utility.js [hunk 1]
```diff
@@ -5,13 +5,16 @@ var Cartesian3 = Cesium.Cartesian3;
 var CesiumMath = Cesium.Math;
 var Matrix3 = Cesium.Matrix3;
 var Matrix4 = Cesium.Matrix4;
+var Plane = Cesium.Plane;
+var Sphere = Cesium.BoundingSphere;
```
Comment (lilleyse) [inline validator/lib/utility.js]: Name `Sphere` to `BoundingSphere` instead.

File: validator/lib/utility.js [hunk 2]
```diff
@@ -142,3 +129,64 @@ function boxInsideSphere(box, sphere) {
...
+    Cartesian3.subtract(point2, point1, a);
+    Cartesian3.subtract(point3, point2, b);
+    Cartesian3.cross(a, b, c);
+    Cartesian3.normalize(c, normal);
+
+    var plane = new Plane.fromPointNormal(point1, normal);
+    return plane;
+}
+
+function createSampleCube() {
```
Comment (lilleyse) [inline validator/lib/utility.js]: Maybe rename to `createUnitCube`?

File: validator/specs/lib/validateTilesetSpec.js [hunk 3]
```diff
@@ -241,6 +241,82 @@ describe('validateTileset', function() {
...
+            sphere: [10, 10, 10, 1]
+        };
+        var tileset = createSampleTileset(tileBoundingVolume, contentBoundingVolume);
+        expect(validateTileset(tileset)
+            .then(function(message) {
+                expect(message).toBe('content sphere [' + contentBoundingVolume.sphere + '] is not within tile box [' + tileBoundingVolume.box + ']');
+          }), done).toResolve();
+    });
+
+    it('returns error message when content\'s center is within tile\'s boundingVolume, but content\'s sphere type boundingVolume is not', function(done) {
```
Comment (lilleyse) [inline validator/specs/lib/validateTilesetSpec.js]: This description could be clearer, maybe: `returns error message when content bounding sphere center is within the tile's bounding box but its radius extends beyond`.

File: validator/specs/lib/validateTilesetSpec.js [hunk 4]
```diff
@@ -241,6 +241,82 @@ describe('validateTileset', function() {
...
+            sphere: [0, 0, 0, 1.1]
+        };
+        var tileset = createSampleTileset(tileBoundingVolume, contentBoundingVolume);
+        expect(validateTileset(tileset)
+            .then(function(message) {
+                expect(message).toBe('content sphere [' + contentBoundingVolume.sphere + '] is not within tile box [' + tileBoundingVolume.box + ']');
+          }), done).toResolve();
+    });
+
+    it('returns error message when content\'s center is outside tile\'s box type boundingVolume, and content\'s sphere type boundingVolume is partially inside tile\'s boundingVolume', function(done) {
```
Comment (lilleyse) [inline validator/specs/lib/validateTilesetSpec.js]: Also reword similar to above.

**Comment by rms13** [comment]
@lilleyse Updated!

**Comment by lilleyse** [comment]
Thanks @rms13.

**Pull request merged**